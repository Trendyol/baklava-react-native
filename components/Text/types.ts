export type TextVariantTypes =
  | 'heading1'
  | 'heading2'
  | 'heading3'
  | 'subtitle01Regular'
  | 'subtitle01Medium'
  | 'subtitle01Semibold'
  | 'subtitle01Bold'
  | 'subtitle02Regular'
  | 'subtitle02Medium'
  | 'subtitle02Semibold'
  | 'subtitle02Bold'
  | 'subtitle03Regular'
  | 'subtitle03Medium'
  | 'subtitle03Semibold'
  | 'subtitle03Bold'
  | 'subtitle04Regular'
  | 'subtitle04Medium'
  | 'subtitle04Semibold'
  | 'subtitle04Bold'
  | 'bodyText'
  | 'bodyUnderline'
  | 'bodyTextLink'
  | 'bodyLongText'
  | 'captionText'
  | 'captionMedium'
  | 'captionLongText'
  | 'captionTextLink';
