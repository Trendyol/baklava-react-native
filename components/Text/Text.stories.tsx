import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from '../Text/Text';
import theme from '../../src/theme';

const variantList = Object.keys(theme.textStyle);

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radios' },
    },
  },
  args: {
    variant: variantList[0] as string,
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Basic: TextStory = args => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Text
    </Text>
    <Text variant={args.variant} p={3}>
      Sample Text
    </Text>
  </>
);

export const Heading: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Heading
    </Text>
    <Text variant="heading1" p={3}>
      Heading 1
    </Text>
    <Text variant="heading2" p={3}>
      Heading 2
    </Text>
    <Text variant="heading3" p={3}>
      Heading 3
    </Text>
  </>
);

export const Subtitle01: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle01Regular" p={3}>
      Subtitle 01 / Regular
    </Text>
    <Text variant="subtitle01Medium" p={3}>
      Subtitle 01 / Medium
    </Text>
    <Text variant="subtitle01Semibold" p={3}>
      Subtitle 01 / Semibold
    </Text>
    <Text variant="subtitle01Bold" p={3}>
      Subtitle 01 / Bold
    </Text>
  </>
);

export const Subtitle02: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle02Regular" p={3}>
      Subtitle 02 / Regular
    </Text>
    <Text variant="subtitle02Medium" p={3}>
      Subtitle 02 / Medium
    </Text>
    <Text variant="subtitle02Semibold" p={3}>
      Subtitle 02 / Semibold
    </Text>
    <Text variant="subtitle02Bold" p={3}>
      Sutitle 02 / Bold
    </Text>
  </>
);

export const Subtitle03: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle03Regular" p={3}>
      Subtitle 03 / Regular
    </Text>
    <Text variant="subtitle03Medium" p={3}>
      Subtitle 03 / Medium
    </Text>
    <Text variant="subtitle03Semibold" p={3}>
      Subtitle 03 / Semibold
    </Text>
    <Text variant="subtitle03Bold" p={3}>
      Subtitle 03 / Bold
    </Text>
  </>
);

export const Subtitle04: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Subtitle
    </Text>
    <Text variant="subtitle04Regular" p={3}>
      Subtitle 04 / Regular
    </Text>
    <Text variant="subtitle04Medium" p={3}>
      Subtitle 04 / Medium
    </Text>
    <Text variant="subtitle04Semibold" p={3}>
      Subtitle 04 / Semibold
    </Text>
    <Text variant="subtitle04Bold" p={3}>
      Subtitle 04 / Bold
    </Text>
  </>
);

export const Body: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Body
    </Text>
    <Text variant="bodyText" p={3}>
      Body / Text
    </Text>
    <Text variant="bodyUnderline" p={3}>
      Body / Underline
    </Text>
    <Text variant="bodyTextLink" p={3}>
      Body / Text Link
    </Text>
    <Text variant="bodyLongText" p={3}>
      Body / Long Text
    </Text>
  </>
);

export const Caption: TextStory = () => (
  <>
    <Text p={3} variant="subtitle01Bold">
      Caption
    </Text>
    <Text variant="captionText" p={3}>
      Caption / Text
    </Text>
    <Text variant="captionMedium" p={3}>
      Caption / Medium
    </Text>
    <Text variant="captionLongText" p={3}>
      Caption / Long Text
    </Text>
    <Text variant="captionTextLink" p={3}>
      Caption / Text Link
    </Text>
  </>
);
