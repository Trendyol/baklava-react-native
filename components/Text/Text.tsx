import React from 'react';
import { Text as T } from 'react-native';
import styled from 'styled-components/native';
import {
  compose,
  color as styledColor,
  space,
  typography,
  position,
  size,
  fontFamily,
  fontWeight,
  textStyle,
  variant as styledVariant,
  fontSize,
} from 'styled-system';
import theme from '../../src/theme';
import { TextVariantTypes } from './types';

const StyledText = styled(T)(
  compose(
    styledColor,
    space,
    typography,
    position,
    size,
    fontSize,
    fontWeight,
    fontFamily,
    textStyle,
    styledVariant({ variants: theme.textStyle }),
  ),
);

function Text({
  variant = 'bodyText',
  color = 'contentPrimary',
  ...rest
}: {
  variant?: TextVariantTypes;
  color?: string;
  [key: string]: any;
}) {
  return (
    <StyledText
      //@ts-ignore
      variant={variant}
      color={color}
      {...rest}>
      {rest.children}
    </StyledText>
  );
}

export default Text;
