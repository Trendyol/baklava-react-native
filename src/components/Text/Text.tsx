import { createText } from '@shopify/restyle';
import { Theme } from '../../theme';

const Text = createText<Theme>();

export default Text;
