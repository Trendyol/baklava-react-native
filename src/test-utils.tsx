import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
