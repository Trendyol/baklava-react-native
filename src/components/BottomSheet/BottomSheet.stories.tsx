import { ComponentMeta, ComponentStory } from '@storybook/react-native';
import React from 'react';
import Text from '../Text/Text';
import BottomSheet from './BottomSheet';
import Button from '../Button/Button';
import Box from '../Box/Box';
import Image from '../Image/Image';
import { useWindowDimensions } from 'react-native';

const BottomSheetMeta: ComponentMeta<typeof BottomSheet> = {
  title: 'BottomSheet',
  component: BottomSheet,
};

export default BottomSheetMeta;

type BottomSheetStory = ComponentStory<typeof BottomSheet>;

export const Basic: BottomSheetStory = () => {
  const dims = useWindowDimensions();

  const [modal1, showModal1] = React.useState(false);
  const [modal2, showModal2] = React.useState(false);

  return (
    <Box borderRadius="l" p="m">
      <Text variant="subtitle1Bold" mb="l">
        BottomSheetStory
      </Text>

      <Box>
        <Button mb="m" filled label="Basic" onPress={() => showModal1(true)} />
        <BottomSheet
          modalKey="basic"
          selectButtonLabel="Select"
          closeButtonLabel="Close"
          visible={modal1}
          title="Basic"
          closeButtonAction={() => showModal1(false)}
          selectButtonAction={() => showModal1(false)}>
          <Text>Basic Content</Text>
        </BottomSheet>
      </Box>

      <Box>
        <Button
          mb="m"
          filled
          label="Basic with image"
          onPress={() => showModal2(true)}
        />
        <BottomSheet
          modalKey="basic"
          selectButtonLabel="Select"
          closeButtonLabel="Close"
          visible={modal2}
          title="Basic with image"
          closeButtonAction={() => showModal2(false)}
          selectButtonAction={() => showModal2(false)}>
          <Image
            height={dims.height * 0.6}
            width={dims.width - 72}
            resizeMode="cover"
            borderRadius={8}
            source={{
              uri: 'https://cdn.dsmcdn.com/ty1006/product/media/images/prod/PIM/20230927/18/a885ced6-7ac8-4ccf-8b58-108db99ed477/1_org_zoom.jpg',
            }}
          />
        </BottomSheet>
      </Box>
    </Box>
  );
};
