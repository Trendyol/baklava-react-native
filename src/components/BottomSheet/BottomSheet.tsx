import React from 'react';
import { TouchableOpacity, StyleSheet } from 'react-native';
import { ModalProps } from '../Modal/Modal';
import Box from '../Box/Box';
import Text from '../Text/Text';
import BaseModal, { A11yProps } from '../Modal/BaseModal';

export type BottomSheetProps = ModalProps & {
  closeButtonLabel: string;
  selectButtonLabel: string;
  selectButtonAction?: () => void;
  title: string;
  header?: React.ReactNode;
};
export const BottomSheetTitle = ({
  closeButtonAction,
  closeButtonLabel,
  selectButtonLabel,
  selectButtonAction,
  testID,
  accessibilityLabel,
  children,
}: {
  closeButtonAction?: () => void;
  closeButtonLabel: string;
  selectButtonLabel: string;
  selectButtonAction?: () => void;
  testID?: string;
  accessibilityLabel?: string;
  children: React.ReactNode;
}) => {
  const testProps = React.useMemo(() => {
    return {
      accessibilityLabel: accessibilityLabel ?? testID,
      testID: testID ?? accessibilityLabel,
    } as A11yProps;
  }, [testID, accessibilityLabel]);

  return (
    <Box flexDirection="column" width="100%" gap="s" mb="m">
      <Box
        justifyContent="space-between"
        width="100%"
        alignItems="center"
        flexDirection="row">
        <TouchableOpacity
          onPress={closeButtonAction}
          style={styles.titleSmallButton}
          testID={`${testProps.testID}-header-close`}
          accessibilityLabel={`${testProps.accessibilityLabel}-header-close`}>
          <Text variant="subtitle3Medium" textAlign="left">
            {closeButtonLabel}
          </Text>
        </TouchableOpacity>
        <Text
          variant="subtitle3Medium"
          textAlign="center"
          style={styles.titleLabel}
          testID={`${testProps.testID}-header-label`}
          accessibilityLabel={`${testProps.accessibilityLabel}-header-label`}
          numberOfLines={1}>
          {children}
        </Text>
        <TouchableOpacity
          onPress={selectButtonAction}
          style={styles.titleSmallButton}
          testID={`${testProps.testID}-header-select`}
          accessibilityLabel={`${testProps.accessibilityLabel}-header-select`}>
          <Text
            variant="subtitle3Medium"
            color="primaryColor"
            textAlign="right">
            {selectButtonLabel}
          </Text>
        </TouchableOpacity>
      </Box>
      <Box height={1} backgroundColor="borderColor" mx="-m" />
    </Box>
  );
};
const BottomSheet = ({
  children,
  closeButtonLabel,
  closeButtonAction,
  selectButtonAction,
  selectButtonLabel,
  accessibilityLabel,
  testID,
  title,
  header,
  ...etc
}: BottomSheetProps) => {
  return (
    <BaseModal
      {...etc}
      justifyContent="flex-end"
      p="none"
      borderBottomStartRadius="none"
      borderBottomEndRadius="none"
      closeButtonAction={closeButtonAction}
      accessibilityLabel={accessibilityLabel}
      testID={testID}
      header={
        header ?? (
          <BottomSheetTitle
            closeButtonLabel={closeButtonLabel}
            closeButtonAction={closeButtonAction}
            selectButtonAction={selectButtonAction}
            selectButtonLabel={selectButtonLabel}
            accessibilityLabel={accessibilityLabel}
            testID={testID}>
            {title}
          </BottomSheetTitle>
        )
      }>
      {children}
    </BaseModal>
  );
};

export default BottomSheet;

const styles = StyleSheet.create({
  titleSmallButton: { width: '25%' },
  titleLabel: { width: '50%' },
});
