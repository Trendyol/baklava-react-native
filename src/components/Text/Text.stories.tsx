import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from './Text';
import theme from '../../theme';

const variantList = Object.keys(theme.textVariants);

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radio' },
    },
  },
  args: {
    variant: variantList[0] as any,
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Basic: TextStory = args => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text
    </Text>
    <Text variant={args.variant} p="2xs">
      Sample Text
    </Text>
  </>
);

export const Heading: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Heading
    </Text>
    <Text variant="heading1" p="2xs">
      Heading 1
    </Text>
    <Text variant="heading2" p="2xs">
      Heading 2
    </Text>
    <Text variant="heading3" p="2xs">
      Heading 3
    </Text>
  </>
);

export const Subtitle1: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 1
    </Text>
    <Text variant="subtitle1Regular" p="2xs">
      Subtitle 1 / Regular
    </Text>
    <Text variant="subtitle1Medium" p="2xs">
      Subtitle 1 / Medium
    </Text>
    <Text variant="subtitle1Semibold" p="2xs">
      Subtitle 1 / Semibold
    </Text>
    <Text variant="subtitle1Bold" p="2xs">
      Subtitle 1 / Bold
    </Text>
  </>
);

export const Subtitle2: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 2
    </Text>
    <Text variant="subtitle2Regular" p="2xs">
      Subtitle 2 / Regular
    </Text>
    <Text variant="subtitle2Medium" p="2xs">
      Subtitle 2 / Medium
    </Text>
    <Text variant="subtitle2Semibold" p="2xs">
      Subtitle 2 / Semibold
    </Text>
    <Text variant="subtitle2Bold" p="2xs">
      Subtitle 2 / Bold
    </Text>
  </>
);

export const Subtitle3: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 3
    </Text>
    <Text variant="subtitle3Regular" p="2xs">
      Subtitle 3 / Regular
    </Text>
    <Text variant="subtitle3Medium" p="2xs">
      Subtitle 3 / Medium
    </Text>
    <Text variant="subtitle3Semibold" p="2xs">
      Subtitle 3 / Semibold
    </Text>
    <Text variant="subtitle3Bold" p="2xs">
      Subtitle 3 / Bold
    </Text>
  </>
);

export const Subtitle4: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 4
    </Text>
    <Text variant="subtitle4Regular" p="2xs">
      Subtitle 4 / Regular
    </Text>
    <Text variant="subtitle4Medium" p="2xs">
      Subtitle 4 / Medium
    </Text>
    <Text variant="subtitle4Semibold" p="2xs">
      Subtitle 4 / Semibold
    </Text>
    <Text variant="subtitle4Bold" p="2xs">
      Subtitle 4 / Bold
    </Text>
  </>
);

export const Body: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Body
    </Text>
    <Text variant="body1" p="2xs">
      Body 1
    </Text>
    <Text variant="body2" p="2xs">
      Body 2
    </Text>
    <Text variant="body3" p="2xs">
      Body 3
    </Text>
    {/* <Text variant="bodyUnderline" p="2xs">
      Body / Underline
    </Text>
    <Text variant="bodyTextLink" p="2xs">
      Body / Text Link
    </Text>
    <Text variant="bodyLongText" p="2xs">
      Body / Long Text
    </Text> */}
  </>
);

export const Caption: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Caption
    </Text>
    <Text variant="captionText" p="2xs">
      Caption / Text
    </Text>
  </>
);
