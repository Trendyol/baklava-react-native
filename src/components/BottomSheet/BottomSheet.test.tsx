import React from 'react';
import { fireEvent, render } from '../../test-utils';
import BottomSheet from './BottomSheet';
import Text from '../Text/Text';

describe('BottomSheet', () => {
  test('should render BottomSheet when visible false', () => {
    // when
    const { toJSON, queryByTestId } = render(
      <BottomSheet
        testID="modal"
        visible={false}
        title="Base Title"
        closeButtonLabel="Close"
        selectButtonLabel="Select"
      />,
    );

    // then
    const modalWrapper = queryByTestId('modal-wrapper');
    expect(modalWrapper).toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render BottomSheet when visible true', () => {
    // when
    const { toJSON, queryByTestId } = render(
      <BottomSheet
        testID="modal"
        visible={true}
        title="Base Title"
        closeButtonLabel="Close"
        selectButtonLabel="Select">
        <Text>Base Content</Text>
      </BottomSheet>,
    );

    // then
    const modalWrapper = queryByTestId('modal-wrapper');
    expect(modalWrapper).not.toBeNull();
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render BottomSheet when closeButtonAction is called', () => {
    // when
    const closeButtonAction = jest.fn();
    const { queryByTestId } = render(
      <BottomSheet
        testID="modal"
        visible={true}
        title="Base Title"
        closeButtonAction={closeButtonAction}
        closeButtonLabel="Close"
        selectButtonLabel="Select">
        <Text>Base Content</Text>
      </BottomSheet>,
    );

    fireEvent.press(queryByTestId('modal-header-close'));

    // then
    expect(closeButtonAction).toHaveBeenCalled();
  });

  test('should render BottomSheet when selectButtonAction is called', () => {
    // when
    const selectButtonAction = jest.fn();
    const { queryByTestId } = render(
      <BottomSheet
        testID="modal"
        visible={true}
        title="Base Title"
        selectButtonAction={selectButtonAction}
        closeButtonLabel="Close"
        selectButtonLabel="Select">
        <Text>Base Content</Text>
      </BottomSheet>,
    );

    fireEvent.press(queryByTestId('modal-header-select'));

    // then
    expect(selectButtonAction).toHaveBeenCalled();
  });

  test('should render BottomSheet when title is given', () => {
    // when
    const { queryByTestId } = render(
      <BottomSheet
        testID="modal"
        visible={true}
        title="Base Title"
        closeButtonLabel="Close"
        selectButtonLabel="Select">
        <Text>Base Content</Text>
      </BottomSheet>,
    );

    // then
    expect(queryByTestId('modal-header-label').children[0]).toEqual(
      'Base Title',
    );
  });
});
