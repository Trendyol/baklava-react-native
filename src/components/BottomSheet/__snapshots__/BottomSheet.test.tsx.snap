// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet should render BottomSheet when visible false 1`] = `null`;

exports[`BottomSheet should render BottomSheet when visible true 1`] = `
<View
  accessibilityLabel="modal-wrapper"
  accessible={true}
  style={
    [
      {
        "bottom": 0,
        "justifyContent": "flex-end",
        "left": 0,
        "padding": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
    ]
  }
  testID="modal-wrapper"
>
  <View
    accessibilityLabel="modal-overlay"
    collapsable={false}
    onResponderRelease={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#273142",
        "bottom": 0,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    testID="modal-overlay"
  />
  <View
    accessibilityLabel="modal"
    accessible={true}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomEndRadius": 0,
          "borderBottomStartRadius": 0,
          "borderRadius": 8,
          "padding": 16,
        },
        {
          "maxHeight": "90%",
        },
      ]
    }
    testID="modal"
  >
    <View
      accessible={false}
      style={
        [
          {
            "flexDirection": "column",
            "gap": 14,
            "marginBottom": 16,
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessible={false}
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "100%",
            },
          ]
        }
      >
        <View
          accessibilityLabel="modal-header-close"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="modal-header-close"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#273142",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "left",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
        <Text
          accessibilityLabel="modal-header-label"
          accessible={true}
          numberOfLines={1}
          style={
            [
              {
                "color": "#273142",
                "fontFamily": "Rubik-Medium",
                "fontSize": 14,
                "fontWeight": "500",
                "lineHeight": 16,
                "textAlign": "center",
              },
              {
                "width": "50%",
              },
            ]
          }
          testID="modal-header-label"
        >
          Base Title
        </Text>
        <View
          accessibilityLabel="modal-header-select"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
              "width": "25%",
            }
          }
          testID="modal-header-select"
        >
          <Text
            accessible={false}
            style={
              [
                {
                  "color": "#F27A1A",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "lineHeight": 16,
                  "textAlign": "right",
                },
              ]
            }
          >
            Select
          </Text>
        </View>
      </View>
      <View
        accessible={false}
        style={
          [
            {
              "backgroundColor": "#D5D9E1",
              "height": 1,
              "marginHorizontal": -16,
            },
          ]
        }
      />
    </View>
    <RCTScrollView
      accessibilityLabel="modal-body"
      bounces={false}
      showsVerticalScrollIndicator={true}
      testID="modal-body"
    >
      <View>
        <Text
          accessible={false}
          style={
            [
              {
                "color": "#273142",
                "fontFamily": "Rubik-Regular",
                "fontSize": 14,
                "textAlign": "left",
              },
            ]
          }
        >
          Base Content
        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
